app/src/main/java/org/thoughtcrime/securesms/conversation/v2/ConversationViewModel.kt

```diff
@@ -32,7 +32,7 @@ class ConversationViewModel(
 ) : ViewModel() {
 
     val showSendAfterApprovalText: Boolean
-        get() = recipient?.run { address.isGroup.not() && hasApprovedMe().not() } ?: false
+        get() = recipient?.run { isContactRecipient && address.serialize() != storage.getUserPublicKey() && !hasApprovedMe() } ?: false
```

Comment:
I think there is also a way to check the recipient is the local number without needing the address, `Recipient.isLocalNumber()` although the naming is carried over from Signal I believe, making it a bit unintuitive